compiler/z/codegen/BinaryCommutativeAnalyser.cpp

```diff
@@ -245,8 +245,12 @@ TR_S390BinaryCommutativeAnalyser::genericAnalyser(TR::Node * root, TR::InstOpCod
 
    bool isLoadNodeNested = false;
 
+   // All of the memToRegOpCodes (MGH, MSGC and MSC) used in the if block below
+   // rely on the miscellaneous-instruction-extension facility 2 being installed
+   const bool supportsMIE2 = cg()->comp()->target().cpu.getSupportsMiscellaneousInstructionExtensions2Facility();
+
    // TODO: add MH and MHY here; outside of the z14 if check.
-   if(cg()->comp()->target().cpu.getSupportsArch(TR::CPU::z14))
+   if(cg()->comp()->target().cpu.getSupportsArch(TR::CPU::z14) && supportsMIE2)
```

Comment:
This shouldn't be an `&&` expression. If we have Misc. Instruction Ext. Facility 2 then we can execute the `MSC` instruction irregardless of what processor we report. We should only guard the code with the CPU check. We can also fold that variable into the `if` since there is only one use of it.

Reply:
> then we can execute the MSC instruction irregardless of what processor we report.

In this case, should we guard the `if` with the facility check and not the CPU check? The reason I ask is because all of the opcodes that can be emitted in this `if` (`MSC`,`MSGC` and `MGH`) require Misc. Instruction Ext. Facility 2. The whole `if` doesn't really do anything if the facility isn't present.

Reply:
> In this case, should we guard the `if` with the facility check and not the CPU check?

Yes. We only need facility checks.

Reply:
Fixed this with the new commit `5486f7c`